mailpoet/lib/Util/Notices/BlackFridayNotice.php

```diff
@@ -22,8 +22,8 @@ public function __construct(
 
   public function init($shouldDisplay) {
     $shouldDisplay = $shouldDisplay
-      && (time() <= strtotime('2021-11-30 12:00:00'))
-      && (time() >= strtotime('2021-11-24 12:00:00'))
+      && (time() >= strtotime('2022-10-03 12:00:00'))
+      && (time() <= strtotime('2022-10-07 12:00:00'))
```

Comment:
I've always been bad when it comes to thinking about time and timezones, so forgive me in advance if this comment doesn't make sense and while I think through it all in this comment.

Should we include UTC in the `strtotime` calls? If we don't then PHP will [default to using whatever the default timezone is](https://www.php.net/manual/en/function.strtotime.php), right? If I had my PHP's default timezone set to UTC-5, then this would evaluate to 

```php
strtotime('2022-10-07 12:00:00 -5:00')
> 1665162000
```

Whereas with UTC we get
```php
strtotime('2022-10-07 12:00:00 UTC')
> 1665144000
```

Since `time()` will always return UTC timestamps isn't it possible we would end up showing this notice after it's valid?

`time() <= strtotime('2022-10-07 12:00:00 America/Chicago')` should be equivalent to `time() <= strtotime('2022-10-07 17:00:00 UTC')`, so I think I could end up seeing this notice for 3 hours after the deal expires (again, only if my default PHP timezone were different -- strtotime doesn't care about the WP timezone).

Reply:
@johnolek for some reason I thought it's harder to work with timezones in PHP, so that's why I suggested a timezone-less approach with 12 hours of reserve on both ends. While officially the sale is from and to 2 pm UTC, the shop coupons operate on days, not hours, so the discount will be applied from midnight to midnight UTC, which should pretty much work out so everyone seeing the banner should be able to purchase it.

Nevertheless, I like your suggestion, so I changed it so it's visible during the official sale period from and to 2 pm UTC. Thanks!

(I made the change instead of Rosta, because I only now got the tracking parameters, which I added)

Reply:
Ahh, that makes sense. Somehow I missed/didn't realize that shop coupons are day based. Thanks for the explanation and the change!